### ISSUE TITLE
Does not understand quoting

### ISSUE THREAD

**Issue by eero-t** [koalaman/shellcheck#2415]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2057
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2 and online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
$ cat test.sh 
#!/bin/sh
if [ \! -x "$1" ]; then
	echo "Not executable"
fi
```

#### Here's what shellcheck currently says:

```sh
$ shellcheck test.sh 

In test.sh line 2:
if [ \! -x "$1" ]; then
        ^-- SC2057: Unknown binary operator.

For more information:
  https://www.shellcheck.net/wiki/SC2057 -- Unknown binary operator.
```

#### Here's what I wanted or expected to see:

Shellcheck not being that confused by quoting.

**Comment by slycordinator** [comment]
It doesn't complain, though, if you use:
`#!/bin/sh
if [ ! -x "$1" ]; then
	echo "Not executable"
fi`

It's not seeming to be confused by the quoting, but by having the not (!) be escaped.

**Comment by TinCanTech** [comment]
```
#!/bin/sh if [ ! -x "$1" ]; then echo "Not executable" fi
^-- SC2096 (error): On most OS, shebangs can only specify a single parameter.
```
Are you using an old version ?

**Comment by eero-t** [comment]
> Are you using an old version ?

What @slycordinator used is not relevant for this issue (I guess he just removed newlines from his example for brevity).  As I mentioned in the original comment, I tested the bug that I reported, also with the online checker.  The issue is with shellcheck not understanding escaping (as you can easily test yourself by using example in original comment).

**Comment by TinCanTech** [comment]
Making changes _for brevity_, which alter the meaning of code, is a terrible policy.

Anyway, your first example does indeed throw this unrelated error and the wiki for 2057 even states to raise a bug report for related errors.

**Comment by slycordinator** [comment]
GitHub removed the newlines.

I put pasted the code into a GitHub code comment block, which includes the
newlines, and then they wouldn't show up. I was going to go back and figure
out how to fix it to display again, but forgot about it after having a long
day at work.

**Comment by slycordinator** [comment]
My comment was to draw light to the fact that it doesn't understand the
escape, where at the time the issue said that it doesn't understand quoting.

**Comment by eero-t** [comment]
> My comment was to draw light to the fact that it doesn't understand the escape, where at the time the issue said that it doesn't understand quoting.

That was just my bad English, I actually had meant escaping. :-)
